fee_allocator/accounting/recon.py

```diff
@@ -10,77 +10,69 @@
-                    x["bal_incentives"],
-                ]
-            )
-            for x in fees.values()
-        ]
-    ), 4)
+    all_incentives_sum = round(
+        sum([sum([x["fees_to_vebal"], x["fees_to_dao"], x["aura_incentives"], x["bal_incentives"], ]) for x in
+             fees.values()]), 4)
     # If everything is 0 - don't store the summary
     if all_incentives_sum == 0 or all_fees_sum == 0:
         return
-    aura_incentives = sum([x["aura_incentives"] for x in fees.values()])
-    bal_incentives = sum([x["bal_incentives"] for x in fees.values()])
-    fees_to_dao = sum([x["fees_to_dao"] for x in fees.values()])
-    fees_to_vebal = sum([x["fees_to_vebal"] for x in fees.values()])
     delta = all_fees_sum - all_incentives_sum
-
     abs_delta = abs(delta)
+
+    aura_incentives = 0
+    bal_incentives = 0
+    fees_to_dao = 0
+    fees_to_vebal = 0
+    # Sum up all incentives and verify there are no negative numbers.
+    for fee_info in fees.values():
+        assert fee_info[
+                   "aura_incentives"] >= 0, f"Recon Failed: {fee_info['pool_id???']} Aura incentives of {fee_info['aura_incentives']}should be >= 0"
+        aura_incentives += fee_info["aura_incentives"]
+        assert fee_info[
+                   "bal_incentives"] >= 0, f"Recon Failed: {fee_info['pool_id???']} Balancer incentives of {fee_info['bal_incentives']}should be >= 0"
+        bal_incentives += fee_info["bal_incentives"]
+        assert fee_info[
+                   "fees_to_dao"] >= 0, f"Recon Failed: {fee_info['pool_id???']} Balancer incentives of {fee_info['fees_to_dao']}should be >= 0"
+        fees_to_dao += fee_info["fees_to_dao"]
+        assert fee_info[
+                   "fees_to_vebal"] >= 0, f"Recon Failed: {fee_info['pool_id???']} Balancer incentives of {fee_info['fees_to_vebal']}should be >= 0"
+        fees_to_vebal += fee_info["fees_to_vebal"]
+        assert fee_info[
+                   "total_incentives"] >= 0, f"Recon Failed: {fee_info['pool_id???']} Balancer incentives of {fee_info['total_incentives']}should be >= 0"
+
     assert abs_delta < Decimal(0.15), f"Reconciliation failed. Delta: {delta}"
     print(f"During recon found a delta of {delta}")
     # Make sure all SUM(pct) == 1
-    assert (
-        round(
-            aura_incentives / all_incentives_sum
-            + bal_incentives / all_incentives_sum
-            + fees_to_dao / all_incentives_sum
-            + fees_to_vebal / all_incentives_sum,
-            4,
-        )
-        == 1
-    ), "Reconciliation failed. Sum of percentages is not equal to 1"
+    assert (round(
+        aura_incentives / all_incentives_sum + bal_incentives / all_incentives_sum + fees_to_dao / all_incentives_sum + fees_to_vebal / all_incentives_sum,
+        4, ) == 1), "Reconciliation failed. Sum of percentages is not equal to 1"
+    # calvulate aura veBAL share
+    aura_vebal_share = round(aura_incentives / (aura_incentives + bal_incentives), 4)
```

Comment:
calculate the current actual veBAL share, and check it is within 5% of target if provided.